tests/platform_tests/test_cont_warm_reboot.py

```diff
@@ -33,132 +34,168 @@
+        @param dut: The AnsibleHost object of DUT.
+        @param interfaces: DUT's interfaces defined by minigraph
+        """
+        logging.info("Wait %d seconds for all the transceivers to be detected" % MAX_WAIT_TIME_FOR_INTERFACES)
+        pytest_assert(wait_until(MAX_WAIT_TIME_FOR_INTERFACES, 20, check_interface_information, self.duthost, interfaces), \
+            "Not all transceivers are detected or interfaces are up in %d seconds" % MAX_WAIT_TIME_FOR_INTERFACES)
+
+        logging.info("Check transceiver status")
+        check_transceiver_basic(self.duthost, interfaces)
+
+        logging.info("Check LAGs and interfaces status")
+        checks.check_interfaces(self.duthost)
+
+
+    def check_neighbors(self):
+        """
+        Perform a BGP neighborship check.
+        """
+        logging.info("Check BGP neighbors status. Expected state - established")
+        bgp_facts = self.duthost.bgp_facts()['ansible_facts']
+        mg_facts  = self.duthost.minigraph_facts(host=self.duthost.hostname)['ansible_facts']
+
+        for value in bgp_facts['bgp_neighbors'].values():
+            # Verify bgp sessions are established
+            pytest_assert(value['state'] == 'established', "BGP session not established")
+            # Verify locat ASNs in bgp sessions
+            pytest_assert(value['local AS'] == mg_facts['minigraph_bgp_asn'], \
+            "Local ASNs not found in BGP session")
+
+        for v in mg_facts['minigraph_bgp']:
+            # Compare the bgp neighbors name with minigraph bgp neigbhors name
+            pytest_assert(v['name'] == bgp_facts['bgp_neighbors'][v['addr'].lower()]['description'], \
+            "BGP neighbor's name does not match minigraph")
+            # Compare the bgp neighbors ASN with minigraph
+            pytest_assert(v['asn'] == bgp_facts['bgp_neighbors'][v['addr'].lower()]['remote AS'], \
+            "BGP neighbor's ASN does not match minigraph")
+
+
+    def start_cont_warm_reboot(self):
+        """
+        @summary: This test case is to perform continuous warm reboot in a row
+        """
+        asic_type = self.duthost.facts["asic_type"]
+        if asic_type in ["mellanox"]:
+            issu_capability = self.duthost.command("show platform mlnx issu")["stdout"]
+            if "disabled" in issu_capability:
+                pytest.skip("ISSU is not supported on this DUT, skip this test case")
+
+        # Start advancedReboot script on the ptf host to enable continuous I/O
+        advancedReboot = self.get_advanced_reboot(rebootType='warm-reboot', enableContinuousIO=self.enable_continuous_io)
+        thr = threading.Thread(target=advancedReboot.runRebootTestcase)
+        thr.setDaemon(True)
+        thr.start()
+
+        file_template = {
+            'install_list': self.image_list, # this list can be modified at runtime to enable testing different images
+            'location': 'http://100.127.20.23/installer/sonic/broadcom/internal-201811/',
```

Comment:
Why this is hardcoded?

Reply:
This should not be hardcoded, the value of CLI argument `--image_location` is to be used here. Fixed now, thanks.